src/prebid.js

```diff
@@ -631,7 +632,10 @@ $$PREBID_GLOBAL$$.requestBids = (function() {
       global: mergeDeep({}, config.getAnyConfig('ortb2') || {}, ortb2 || {}),
       bidder: Object.fromEntries(Object.entries(config.getBidderConfig()).map(([bidder, cfg]) => [bidder, cfg.ortb2]).filter(([_, ortb2]) => ortb2 != null))
     }
-    return startAuction({bidsBackHandler, timeout: cbTimeout, adUnits, adUnitCodes, labels, auctionId, ttlBuffer, ortb2Fragments, metrics, defer});
+    return enrichFPD(GreedyPromise.resolve(ortb2Fragments.global)).then(global => {
+      ortb2Fragments.global = global;
```

Comment:
This line here is never hit

Reply:
ok the diff between a page that `works` and one that does not is 

`ortb2Fragments.global`

Woks:
```
{
    "global": {
        "regs": {
            "ext": {
                "us_privacy": "1YNY"
            }
        }
    },
    "bidder": {}
}
```

Silently Fails:

```
{
    "global": {},
    "bidder": {}
}
```

So, then it seems the `enrichFpd` funciton is not resolving 

```
GreedyPromise.all([fpd, getSUA().catch(() => null)])
```

because `fpd` here is an unfulfilled promise

![image](https://user-images.githubusercontent.com/13972794/222827269-1d46a15f-82b5-461f-92ee-5484fb93cd2a.png)

I'll continue to look into this.

Perhaps https://github.com/prebid/Prebid.js/pull/9558 fixes this, will test now.

Reply:
And it does fix it

Now `fpd` at the line is
![image](https://user-images.githubusercontent.com/13972794/222828640-8d6e287f-5fb1-4c70-a155-873fd05d730f.png)

So somehow this change fixed our issue here.

I am not too knowledgeable on the whole GreedyPromise thing as I should be. So would love some knowledgeable input from you @dgirardi if any of this makes sense from what I put.

Obviously I can explain more if needed.

This enrich function is the part which is never fulfilling (the `fpd` var here is the one I am mentioning above)
```js
export const enrichFPD = hook('sync', (fpd) => {
  return GreedyPromise.all([fpd, getSUA().catch(() => null)])
    .then(([ortb2, sua]) => {
      Object.entries(ENRICHMENTS).forEach(([section, getEnrichments]) => {
        const data = getEnrichments();
        if (data && Object.keys(data).length > 0) {
          ortb2[section] = mergeDeep({}, data, ortb2[section]);
        }
      });
      if (sua) {
        deepSetValue(ortb2, 'device.sua', Object.assign({}, sua, ortb2.device.sua));
      }
      return ortb2;
    });
});
```

Reply:
ok, turns out it is literally because certain pubs using different versions of promise I believe!

Which is mentioned in the PR https://github.com/prebid/Prebid.js/pull/9558

Ok sorry for all the messages here, was using this as my soundboard!